echo.go

```diff
@@ -62,20 +61,28 @@ import (
 
 type (
 	// Echo is the top-level framework instance.
+	//
+	// Goroutine safety: Do not mutate Echo instance fields after server has started. Accessing these
+	// fields from handlers/middlewares and changing field values at the same time leads to data-races.
+	// Same rule applies to adding new routes after server has been started - Adding a route is not Goroutine safe action.
 	Echo struct {
 		filesystem
 		common
 		// startupMutex is mutex to lock Echo instance access during server configuration and startup. Useful for to get
 		// listener address info (on which interface/port was listener binded) without having data races.
-		startupMutex     sync.RWMutex
+		startupMutex sync.RWMutex
+		colorer      *color.Color
+
+		// premiddleware are middlewares that are run before routing is done. In case pre-middleware returns an error router
+		// will not be called at all and execution ends up in global error handler.
```

Comment:
Suggestion:

```diff
-   // premiddleware are middlewares that are run before routing is done. In case pre-middleware returns an error router
-		// will not be called at all and execution ends up in global error handler.
+   // premiddleware are middlewares that are run before routing is done. In case a pre-middleware returns 
+   // an error the router is not executed and the request will end up in the global error handler
```